vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -397,6 +397,30 @@ public VimeoAccount authorizeWithClientCredentialsGrantSync() {
         return call;
     }
 
+    /**
+     * This method is used to exchange a user access token from app A for a token for app B.
+     * <p/>
+     * An app can use this to exchange an authenticated user's access token from app A for a new access
+     * token for app B, using the scope associated with app B. The caller of this method will be app B.
+     * <p/>
+     * <b>This method is application restricted, and not meant to be consumed by the general public.</b>
+     *
+     * @param token    the authenticated user access token from application A. This <b>cannot</b> be a client
+     *                 credentials access token.
+     * @param callback Callback invoked upon success/fail of the service
+     * @return A Call object
+     */
+    public Call<VimeoAccount> singleSignOnTokenExchange(@NotNull String token, AuthCallback callback) {
+        if (callback == null) {
+            throw new AssertionError("Callback cannot be null");
+        }
+
+        Call<VimeoAccount> call =
+                mVimeoService.ssoTokenExchange(getBasicAuthHeader(), token, mConfiguration.scope);
+        call.enqueue(new AccountCallback(this, callback));
+        return call;
+    }
```

Comment:
We haven't officially found a way to prevent people from using a thing, but we could maybe do what the android folks do and throw @hide on it.

http://stackoverflow.com/a/17056643/1759443